### PR TITLE
Upgrade dubbo-go to v3.1.1

### PR DESCRIPTION
**Pull request by mark4z**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/config/xds/apiclient/grpc_envoy.go [hunk 1]
```diff
@@ -19,12 +19,12 @@ package apiclient
 
 import (
 	"context"
+	"github.com/apache/dubbo-go-pixiu/pkg/logger"
```
Comment (AlexStocks) [inline pkg/config/xds/apiclient/grpc_envoy.go]: 这个不怎么 dubbogo 了，哈哈，换下位置吧

Comment (mark4z) [reply pkg/config/xds/apiclient/grpc_envoy.go]: dubbo go 的logger切到gost了

File: go.mod [hunk 2]
```diff
@@ -3,23 +3,23 @@ module github.com/apache/dubbo-go-pixiu
 go 1.18
 
 require (
-	dubbo.apache.org/dubbo-go/v3 v3.0.2-0.20220519062747-f6405fa79d5c
+	dubbo.apache.org/dubbo-go/v3 v3.1.1-rc1
```
Comment (FoghostCn) [inline go.mod]: why rc version but not v3.1.1

Comment (mark4z) [reply go.mod]: I'll fix it.

**Review by baerwang — APPROVED** [review]

**Pull request merged**